be/src/exec/pipeline/scan_operator.cpp

```diff
@@ -48,135 +53,169 @@ bool ScanOperator::has_output() const {
         return false;
     }
 
-    // _chunk_source init at pull_chunk()
-    // so the the initialization of the first chunk_source needs
-    // to be driven by has_output() returning true
-    if (!_chunk_source) {
-        return true;
+    for (const auto& chunk_source : _chunk_sources) {
+        if (chunk_source != nullptr && chunk_source->has_output()) {
+            return true;
+        }
     }
 
-    DCHECK(_chunk_source != nullptr);
+    if (_num_running_io_tasks >= _max_io_tasks_per_op) {
```

Comment:
Do we really need this check？

Reply:
This check can reduce the number of noneffective schedule times. Otherwise, the driver may be moved from blocked poller queue to dispatcher queue, but it cannot trigger a new i/o task.